internal/controller/operatorconfigmap_controller.go

```diff
@@ -507,11 +545,25 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI(storageClients *v1al
 	nfsDriver := &csiopv1a1.Driver{}
 	nfsDriver.Name = templates.NfsDriverName
 	nfsDriver.Namespace = c.OperatorNamespace
-
 	if err := c.createOrUpdate(nfsDriver, func() error {
 		if err := c.own(nfsDriver); err != nil {
 			return fmt.Errorf("failed to own csi nfs driver: %v", err)
 		}
+		// only update during initial creation
+		if nfsDriver.UID == "" {
```

Comment:
Your condition should be based on the UID (as it is here) and the existence of custom tolerations (missing check)